src/whip.js

```diff
@@ -42,12 +42,14 @@ class JanusWhipServer extends EventEmitter {
 			throw new Error('Invalid configuration, unsupported "debug" level');
 		this.config = {
 			janus: {
-				address: janus.address
+				address: janus.address,
+				apisecret: janus.apisecret
 			},
 			rest: {
 				port: rest.port,
 				basePath: rest.basePath,
-				app: rest.app
+				app: rest.app,
+				https: rest.https
```

Comment:
Not sure what this is supposed to do? But if you're adding properties to the constructor that were not there before, they should be validated first, especially if they're expected to be of a specific type. See what we do for existing properties in the lines above.